plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/TargetsPresenter.java

```diff
@@ -700,24 +740,32 @@ protected void onErrorReceived(Throwable exception) {
      * Ensures machine is started.
      */
     private void onConnected(final String machineId) {
-        machineService.getMachine(machineId).then(new Operation<MachineDto>() {
+        // There is a little bug in machine service on the server side.
+        // The machine info is updated with a little delay after running a machine.
+        // Using timer must fix the problem.
+        new Timer() {
```

Comment:
Not clear about timer, how it solve problem? Why you set 500 not 600? Can accept it as urgent fix but must be fixed ASAP. Other ok for me